spec/services/git_based_domain_import_service_spec.rb

```diff
@@ -1,66 +1,87 @@
 describe GitBasedDomainImportService do
-  describe "#import" do
-    let(:git_repo) { double("GitRepository", :git_branches => git_branches, :id => 123) }
-    let(:domain) { double("MiqAeDomain") }
+  let(:git_repo) { double("GitRepository", :git_branches => git_branches, :id => 123) }
+  let(:domain) { double("MiqAeDomain") }
+  let(:user) { FactoryGirl.create(:user) }
+  let(:task) { FactoryGirl.create(:miq_task) }
+  let(:ref_name) { 'the_branch_name' }
+  let(:ref_type) { 'branch' }
+  let(:task_options) { {:action => "Import git repository", :userid => user.userid} }
+  let(:queue_options) do
+    {
+      :class_name  => "MiqAeDomain",
+      :method_name => "import_git_repo",
+      :role        => "git_owner",
+      :args        => [import_options]
+    }
+  end
+  let(:import_options) do
+    {
+      "git_repository_id" => git_repo.id,
+      "ref"               => ref_name,
+      "ref_type"          => ref_type,
+      "tenant_id"         => 321
+    }
+  end
 
+  describe "#import" do
     before do
-      allow(GitRepository).to receive(:find_by).with(:id => 123).and_return(git_repo)
+      allow(GitRepository).to receive(:find_by).with(:id => git_repo.id).and_return(git_repo)
       allow(domain).to receive(:update_attribute).with(:enabled, true)
+      allow(MiqTask).to receive(:wait_for_taskid).with(task.id).and_return(task)
+      allow(task).to receive(:task_results).and_return(domain)
+      User.current_user = user
```

Comment:
This should be another `allow`, I think: `allow(User).to receive(:current_user).and_return(user)`